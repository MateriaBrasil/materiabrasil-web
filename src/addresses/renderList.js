import React, { Fragment } from 'react'

import map from 'lodash/map'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import Avatar from '@material-ui/core/Avatar'
import BuildIcon from '@material-ui/icons/Build'
import EmailIcon from '@material-ui/icons/Email'

import Card from './Card'

export default props => list => {
  return (
    <Fragment>
      {(list && list.length) > 0 ? (
        <Card title="EndereÃ§o">
          <List>
            {map(
              list,
              ({
                id,
                addressType,
                streetAddress,
                zipCode,
                city,
                state,
                country,
              }) => {
                const fullAddress = `${streetAddress}${
                  zipCode ? ` - ${zipCode}` : ''
                }`
                const fullCity = `${city}, ${state}, ${country}`

                return (
                  <ListItem key={id}>
                    <Avatar>
                      {addressType === 'manufacture' ? (
                        <BuildIcon />
                      ) : (
                        <EmailIcon />
                      )}
                    </Avatar>
                    <ListItemText primary={fullAddress} secondary={fullCity} />
                  </ListItem>
                )
              },
            )}
          </List>
        </Card>
      ) : null}
    </Fragment>
  )
}
