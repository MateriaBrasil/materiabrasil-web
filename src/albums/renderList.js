import React from 'react'
import map from 'lodash/map'
import List from '@material-ui/core/List'
import Dialog from 'materials/Dialog'
import renderAlbums from './renderAlbums'

export default props => routeProps => {
  return (
    <Dialog {...props} {...routeProps} title="Adicionar a um Ã¡lbum">
      <List style={{ width: '100%' }}>
        {map(
          props.currentUser && props.currentUser.albums,
          renderAlbums({ ...props, ...routeProps }),
        )}
      </List>
    </Dialog>
  )
}
