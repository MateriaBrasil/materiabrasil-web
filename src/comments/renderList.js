import React from 'react'
import Link from 'react-router-dom/Link'
import map from 'lodash/map'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import Avatar from '@material-ui/core/Avatar'
import PersonIcon from '@material-ui/icons/Person'

export default props => list => (
  <List>
    {map(list, ({ id, user: { name, id: userId }, text }) => (
      <Link to={`/users/${userId}`} style={{ textDecoration: 'none' }} key={id}>
        <ListItem button>
          <Avatar>
            <PersonIcon />
          </Avatar>
          <ListItemText primary={name} secondary={text} />
        </ListItem>
      </Link>
    ))}
  </List>
)
