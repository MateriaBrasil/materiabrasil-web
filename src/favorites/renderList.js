import React from 'react'
import Link from 'react-router-dom/Link'
import GridList from '@material-ui/core/GridList'
import GridListTile from '@material-ui/core/GridListTile'

import TitleBar from '../materials/list/TitleBar'
import Destroy from './Destroy'

export default ({ showDestroy, parentId }) => list => (
  <GridList cellHeight={360}>
    {list.map(({ id, favoritable, destroying }, index) => {
      const { id: favoritableId, name, imageUrl } = favoritable
      const renderIcons =
        showDestroy && (() => <Destroy parentId={parentId} id={id} destroying={destroying} />)

      return (
        <GridListTile key={id}>
          <Link to={`/${favoritableId}`}>
            <img
              src={imageUrl}
              alt={name}
              style={{ width: '100%', objectFit: 'cover', height: 360 }}
            />
          </Link>
          <TitleBar material={favoritable} renderIcons={renderIcons} />
        </GridListTile>
      )
    })}
  </GridList>
)
