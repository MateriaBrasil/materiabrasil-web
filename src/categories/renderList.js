import React from 'react';
import { List } from 'croods';

import renderList from './renderMaterialCategoriesList';

export default props => categories => (
  <List
    name="materialCategories"
    path={`/materials/${props.match.params.id}/categories`}
    render={renderList({ ...props, categories })}
  />
);
