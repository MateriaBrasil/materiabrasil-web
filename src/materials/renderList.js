import React, { Fragment } from 'react'
import { List } from 'croods'
import { Helmet } from 'react-helmet'

import Filters from 'categories/Filters'
import renderList from './list/render'

export default props => routeProps => {
  return (
    <Fragment>
      <Helmet>
        <title>Explore</title>
      </Helmet>
      <Filters
        {...props}
        {...routeProps}
        render={({ categories }) => (
          <List
            name="materials"
            path={`/materials?${categories}`}
            render={renderList(props)}
          />
        )}
      />
    </Fragment>
  )
}
