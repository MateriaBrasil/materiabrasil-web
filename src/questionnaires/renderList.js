import React from 'react'
import { List } from 'croods'
import renderList from './list/render'

export default props => routeProps => {
  const { id } = routeProps.match.params

  return (
    <List
      id={id}
      name="questionnaires"
      path={`/questionnaires?about_type=${props.aboutType}`}
      render={renderList({ ...props, ...routeProps })}
    />
  )
}
