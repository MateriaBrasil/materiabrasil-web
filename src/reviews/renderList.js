import React from 'react'
import { List } from 'croods'

import renderList from './list/render'

export default props => routeProps => {
  const { match } = routeProps
  const { params } = match
  const { id } = params

  return (
    <List
      parentId={id}
      name="reviews"
      path={`/materials/${id}/reviews`}
      render={renderList({ ...props, ...routeProps, id })}
    />
  )
}
