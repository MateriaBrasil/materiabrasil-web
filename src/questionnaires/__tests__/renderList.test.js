import React from 'react'
import renderer from 'react-test-renderer'

import renderList from '../renderList'

jest.mock('croods', () => ({
  List: props => <div {...props}>{props.children}</div>,
}))

jest.mock('../list/render', () => props => list => <div>List</div>)

it('renders correctly', () => {
  const props = { aboutType: 'foo' }
  const routeProps = { match: { params: { id: 123 } } }
  const tree = renderer.create(renderList(props)(routeProps)).toJSON()
  expect(tree).toMatchSnapshot()
})
