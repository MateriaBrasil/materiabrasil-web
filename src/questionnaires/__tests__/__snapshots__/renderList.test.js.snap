// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id={123}
  name="questionnaires"
  path="/questionnaires?about_type=foo"
  render={[Function]}
/>
`;
